src/traces/waterfall/cross_trace_calc.js

```diff
@@ -65,6 +65,11 @@ module.exports = function crossTraceCalc(gd, plotinfo) {
             if(di.isSum === false) {
                 di.s0 += (j === 0) ? 0 : cd[j - 1].s;
             }
+
+            if(j + 1 < cd.length) {
```

Comment:
Non blocking, but I think `j < cd.length - 1` is more readable.

Reply:
I wanted to simply note that `j + 1` result is going to be used inside the `cd` array below.
So if you don't mind we may keep this the way it is.